test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -1046,6 +1051,145 @@ private void testSnapshotSyncAndLogEntrySync(int numCyclesToDelayApply, boolean
     }
 
+     /**
+     * Test Log Entry (delta) Sync for the case where the ACKs are arbitrarily dropped
+     * for a fixed number of times at the Source. This will test that LR is
+     * (i) not impacted by dropped ACKs
+     * (ii) Source resends msgs for which it hasn't received the ACKs
+     * (iii) Sink handles the already seen and processed msgs.
+     **/
+    @Test
+    public void testLogEntrySyncWithAckDrops() throws Exception {
+        setupEnv();
+        // Write data in transaction to t0 and t1
+        Set<String> crossTables = new HashSet<>();
+        crossTables.add(t0);
+        crossTables.add(t1);
+
+        // Writes transactions to t0, t1 and t2 + transactions across 'crossTables'
+        writeCrossTableTransactions(crossTables, true);
+
+        // Start Log Entry Sync
+        expectedAckMessages = NUM_KEYS * WRITE_CYCLES;
+        testConfig.clear().setDropAckLevel(1);
+
+        startLogEntrySync(crossTables, WAIT.ON_ACK);
+
+        // Verify Data on Destination site
+        log.debug("****** Verify Data on Destination");
+        // Because t2 is not specified as a replicated table, we should not see it on the destination
+        srcDataForVerification.get(t2).clear();
+
+        // Verify Destination
+        verifyData(dstCorfuTables, srcDataForVerification);
+    }
+
+
+     /**
+     * Test Log Entry (delta) Sync for the case where messages are arbitrarily dropped at the
+     * destination and ACKs are arbitrarily dropped at the source. This tests
+     * (i) messages dropped at the destination is resent. This would also test that Sink handles out of order messages.
+     * (ii) messages for which ACKs were dropped at the source are resent and Sink handles already processed data.
+     */
+    @Test
+    public void testLogEntrySyncWithMsgDropsAndAckDrops() throws Exception {
+        setupEnv();
+        // Write data in transaction to t0 and t1
+        Set<String> crossTables = new HashSet<>();
+        crossTables.add(t0);
+        crossTables.add(t1);
+
+        writeCrossTableTransactions(crossTables, true);
+
+        // Start Log Entry Sync
+        expectedAckMessages =  NUM_KEYS*WRITE_CYCLES;
+
+        testConfig.clear().setDropMessageLevel(1);
+        testConfig.setDropAckLevel(1);
+
+        startLogEntrySync(crossTables, WAIT.ON_ACK);
+
+        // Verify Data on Destination site
+        log.debug("****** Verify Data on Destination");
+
+        // Because t2 is not specified as a replicated table, we should not see it on the destination
+        srcDataForVerification.get(t2).clear();
+
+        // Verify Destination
+        verifyData(dstCorfuTables, srcDataForVerification);
+    }
+
+    /**
+    * Test Log Entry (delta) Sync for the case where an ACK gets dropped at the Source, and
+    * immediately, the Source gets STOP event followed by log_entry Sync event. This tests that even
+    * though the metadata on Source and destination do not match, the sink is able to process the ms
```

Comment:
Oh no Sink doesn't need to send any msg if LR was stopped on source. 
I meant Active looses an ACK and immediately LR on active stops...say the state transitions on active is as such: STOP LOG_ENTRY SYNC -> INITIALIZED -> LOG_ENTRY SYNC
The pending queue on active that stores entries until it is ACK'ed is now cleared. So now, since active doesn't have any inkling that it had previously sent a msg but not received an ACK, it retains its old metadata, while standby having processed the last msg successfully, has moved ahead.
In the current code, the msg sent by Active (with old unchanged metadata) will be ignored by Standby since Standby thinks that it has already processed it. 

This is an issue since after LR is stopped and started again, there is no guarantee that the msg will have the same log_entries as previously sent (and for which ACK was lost). It can contain new log entries, which should be processed by Standby.

This test case is trying to emulate that scenario.